src/daft-csv/src/read.rs

```diff
@@ -168,29 +243,95 @@ async fn read_csv_single_into_table(
             .unwrap()
     });
     // Collect all chunks in chunk x column form.
-    let chunks = chunk_stream
+    let tables = chunk_stream
         // Limit the number of chunks we have in flight at any given time.
-        .try_buffered(max_chunks_in_flight)
+        .try_buffered(max_chunks_in_flight);
+
+    // Fields expected for read and evaluating predicate
+    let read_fields = if let Some(read_columns) = &read_columns {
+        let field_map = fields
+            .iter()
+            .map(|field| (field.name.clone(), field.clone()))
+            .collect::<HashMap<String, Field>>();
+        read_columns
+            .iter()
+            .map(|col| field_map[col].clone())
+            .collect::<Vec<_>>()
+    } else {
+        fields.clone()
+    };
+
+    // Fields expected as the output. (removing fields that are only needed for predicate evaluation)
+    let schema_fields = if let Some(include_columns) = &include_columns {
+        let field_map = fields
+            .iter()
+            .map(|field| (field.name.clone(), field.clone()))
+            .collect::<HashMap<String, Field>>();
+        include_columns
+            .iter()
+            .map(|col| field_map[col].clone())
+            .collect::<Vec<_>>()
+    } else {
+        fields
+    };
+
+    let read_schema: arrow2::datatypes::Schema = read_fields.clone().into();
+    let read_schema = Arc::new(Schema::try_from(&read_schema)?);
+    let owned_read_schema = read_schema.clone();
+
+    let schema: arrow2::datatypes::Schema = schema_fields.clone().into();
```

Comment:
I don't think that this `.clone()` since `schema_fields` should already be owned and we don't use it below, but lmk if I'm missing something.
```suggestion
    let schema: arrow2::datatypes::Schema = schema_fields.into();
```